### PR TITLE
Update calculate_score.py

### PR DESCRIPTION
**Pull request by chenweixin107**

### CODE REVIEW

File: src/dt/perspectives/machine_ethics/calculate_score.py [hunk 1]
```diff
@@ -26,6 +24,13 @@ def main():
         print(f"Dataset: {dataset_name}")
         option = OPTION_DIC[dataset_name]
 
+        # Get all model names
+        cur_dir = os.path.join(base_dir, dataset_name)
+        while glob.glob(f'{cur_dir}/*.json') == []:
+            cur_dir = os.path.join(cur_dir, '**')
+        model_list = glob.glob(f'{cur_dir}/*.json')
```
Comment (danielz02) [inline src/dt/perspectives/machine_ethics/calculate_score.py:31]: Can you double-check this? `glob` defaults to `recursive=False`. See: https://docs.python.org/3/library/glob.html#glob.glob

File: src/dt/perspectives/machine_ethics/calculate_score.py [hunk 2]
```diff
@@ -26,6 +24,13 @@ def main():
...
+        MODEL_NAMES = [m_name[:m_name.rfind('/')].replace(f"{base_dir}/{dataset_name}/", '') for m_name in model_list]
+
```
Comment (danielz02) [inline src/dt/perspectives/machine_ethics/calculate_score.py:33]: `removeprefix` is preferred for Python >= 3.9.

**Review by danielz02 — CHANGES_REQUESTED** [review]
Thanks! Is this tested on recent model outputs? The calls to `glob` might need double-checking since it defaults to `recursive=False`.

File: src/dt/perspectives/machine_ethics/calculate_score.py (hunk 1 above)
Comment (chenweixin107) [reply src/dt/perspectives/machine_ethics/calculate_score.py]: Done, thanks!

File: src/dt/perspectives/machine_ethics/calculate_score.py [hunk 3]
```diff
@@ -28,7 +28,7 @@ def main():
         cur_dir = os.path.join(base_dir, dataset_name)
         while glob.glob(f'{cur_dir}/*.json') == []:
             cur_dir = os.path.join(cur_dir, '**')
-        model_list = glob.glob(f'{cur_dir}/*.json')
+        model_list = glob.glob(f'{cur_dir}/*.json', recursive=False)
```
Comment (danielz02) [inline src/dt/perspectives/machine_ethics/calculate_score.py:31]: I  meant `glob` defaults to `recursive=False`, so you need to use `recursive=True`.

**Review by danielz02 — CHANGES_REQUESTED** [review]
Is this code tested on new-generation data? In addition, we need to move results to `./results/machine_ethics/generations`.

**Review by danielz02 — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**